packages/oas/src/converter/parts/postdata/BodyConverter.ts

```diff
@@ -0,0 +1,188 @@
+import { Sampler } from '../Sampler';
+import { SubConverter } from '../../SubConverter';
+import { OpenAPI, OpenAPIV3, PostData } from '@har-sdk/core';
+import { toXML, XmlElement } from 'jstoxml';
+import { stringify } from 'qs';
+
+export abstract class BodyConverter<T extends OpenAPI.Document>
+  implements SubConverter<PostData | null>
+{
+  private readonly JPG_IMAGE = '/9j/7g=='; // 0xff, 0xd8, 0xff, 0xee
+  private readonly PNG_IMAGE = 'iVBORw0KGgo='; // 0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0A, 0x1a, 0x0a
+  private readonly ICO_IMAGE = 'AAABAA=='; // 0x00, 0x00, 0x01, 0x00
+  private readonly GIF_IMAGE = 'R0lGODdh'; // 0x47, 0x49, 0x46, 0x38, 0x37, 0x61
+  private readonly BOUNDARY = '956888039105887155673143';
+  private readonly BASE64_PATTERN =
+    /^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$/;
+
+  protected constructor(
+    protected readonly spec: T,
+    protected readonly sampler: Sampler
+  ) {}
+
+  public abstract convert(path: string, method: string): PostData | null;
+
+  protected abstract getContentType(
+    path: string,
+    method: string
+  ): string | undefined;
+
+  protected encodePayload(
+    data: unknown,
+    contentType: string,
+    encoding?: OpenAPIV3.EncodingObject
+  ): { mimeType: string; text: string } {
+    let encodedData = data;
+
+    if (encoding) {
+      encodedData = this.encodeProperties(
+        Object.keys(encoding),
+        data,
+        encoding
+      );
+    }
+
+    return {
+      mimeType: contentType.includes('multipart')
+        ? `${contentType}; boundary=${this.BOUNDARY}`
+        : contentType,
+      text: this.encodeValue(encodedData, contentType)
+    };
+  }
+
+  // eslint-disable-next-line complexity
+  private encodeValue(
+    value: unknown,
+    contentType: string,
+    encoding?: string
+  ): string {
+    const [mime]: string[] = contentType
+      .split(',')
+      .map((x) => x.trim().replace(/;.+?$/, ''));
+
+    switch (mime) {
+      case 'application/json':
+        return this.encodeJson(value);
+      case 'application/x-www-form-urlencoded':
+        return this.encodeFormUrlencoded(value);
+      case 'application/xml':
+        return this.encodeXml(value);
+      case 'multipart/form-data':
+      case 'multipart/mixin':
```

Comment:
```suggestion
      case 'multipart/mixed':
```
also not sure that result of `encodeMultipartFormData()` is valid for "mixed" type as e. g. we have hardcoded `Content-Disposition: form-data`. imo test cases should be introduced if we are going to support it

Reply:
ditto https://github.com/NeuraLegion/har-sdk/pull/176#discussion_r1109383390

Reply:
opened: https://github.com/NeuraLegion/har-sdk/issues/182